frontend/javascripts/dashboard/dataset/dataset_import_view.js

```diff
@@ -162,10 +188,21 @@ class DatasetImportView extends React.PureComponent<Props, State> {
   }
 
   getDatasourceDiffAlert() {
+    // Only show if the option did not apply
+    const {
+      savedDataSourceOnServer,
+      differenceBetweenDatasources,
+      didApplyDifferenceBetweenDatasources,
+      inferredDataSource,
+    } = this.state;
+    const didUserAppliedSuggestions =
+      didApplyDifferenceBetweenDatasources === true &&
+      isDatasourceJSONValid(savedDataSourceOnServer);
     if (
-      this.state.previousDataSource == null ||
-      this.state.previousDataSource.status === notImportedYetStatus ||
-      _.size(this.state.differenceBetweenDatasources) === 0
+      savedDataSourceOnServer == null ||
+      savedDataSourceOnServer.status === notImportedYetStatus ||
+      _.size(differenceBetweenDatasources) === 0 ||
+      didUserAppliedSuggestions
```

Comment:
Shouldn't the diff always be zero when `didUserAppliedSuggestions == true`? Do we need the check for `didUserAppliedSuggestions`, then?